src/js/utils/ServiceValidatorUtil.js

```diff
@@ -17,6 +17,65 @@ const ServiceValidatorUtil = {
         segment === '.' || segment === '..' ||
         serviceIDSegmentPattern.test(segment);
     });
+  },
+
+  /**
+   * Checks if the given server response is a Pod
+   *
+   * @param {object} data - The server response
+   * @returns {boolean} - True if the response refers to a pod
+   */
+  isPodResponse(data) {
+    return typeof data.spec === 'object' &&
+           ServiceValidatorUtil.isPodSpecDefinition(data.spec);
+  },
+
+  /**
+   * Checks if the given definition (commonly form the ServiceForm)
+   * is a Pod specification.
+   *
+   * @param {object} data - The definition
+   * @returns {boolean} - True if the response refers to a pod
+   */
+  isPodSpecDefinition(data) {
+    return Array.isArray(data.containers);
+  },
+
+  /**
+   * Checks if the given server response is an application
+   *
+   * @param {object} data - The server response
+   * @returns {boolean} - True if the response refers to a pod
+   */
+  isApplicationResponse() {
+    // This is a placeholder for proper detection between application and
```

Comment:
Let's create an issue to not forget to implement a proper application detection.